src/parser.ts

```diff
@@ -139,6 +149,12 @@ class Parser {
             const defaultProps = this.extractDefaultPropsFromComponent(exp, source);
             const props = this.getPropsInfo(propsType, defaultProps);
 
+            if (this.opts.ignoreChildrenIfNoDocAvailable &&
```

Comment:
I think it would be better to change the `getPropsInfo` to return list of props including property name. Then the filtering props out would be easier.